crates/keyshare/src/threshold_keyshare.rs

```diff
@@ -330,27 +378,88 @@ impl ThresholdKeyshare {
         Ok(())
     }
 
-    /// 1. CiphernodeSelected
+    pub fn handle_encryption_key_created(
+        &mut self,
+        msg: EncryptionKeyCreated,
+        self_addr: Addr<Self>,
+    ) -> Result<()> {
+        info!("Received EncryptionKeyCreated forwarding to encryption key collector!");
+        let collector = self.ensure_encryption_key_collector(self_addr)?;
+        collector.do_send(msg);
+        Ok(())
+    }
+
+    /// 1. CiphernodeSelected - Generate BFV keys and start collecting
     pub fn handle_ciphernode_selected(
         &mut self,
         msg: CiphernodeSelected,
         address: Addr<Self>,
     ) -> Result<()> {
-        // Ensure the collector is created
         let _ = self.ensure_collector(address.clone());
-        // Initialize State
+        let _ = self.ensure_encryption_key_collector(address.clone());
+
+        let params = get_share_encryption_params();
```

Comment:
Smells to me like this aught to live on self perhaps as it is configuration and is used many times?